nupm/install.nu

```diff
@@ -0,0 +1,43 @@
+use std log
+
+def nupm-home [] {
+    $env.NUPM_HOME? | default (
+        $env.XDG_DATA_HOME?
+        | default ($nu.home-path | path join ".local" "share")
+        | path join "nupm"
+    )
+}
+
+def prepare-directory [directory: path] {
+    rm --recursive --force $directory
+    mkdir $directory
+}
+
+def copy-directory-to [destination: path] {
+    let source = $in
+
+    log info "copying directory"
+    log debug $"source: ($source)"
+    log debug $"destination: ($destination)"
+
+    ls --all $source | where name != ".git" | each {|it|
+        log debug ($it.name | str replace $source "" | str trim --left --char "/")
+        cp --recursive $it.name $destination
+    }
+}
+
+# install a Nushell package
+export def main [
+    --path: path  # the path to the local source of the package (defaults to the current directory)
+] {
+    let path = ($path | default . | path expand)
+
+    # NOTE: here, we suppose that the package file exists and is valid
+    let package = ($path | path join "package.nuon" | open)
```

Comment:
Error handling (non-existent package file, corrupted package file, lack of file permissions etc) here

Reply:
in this PR, with the goal of being simple, i was inspired by [this message](https://discord.com/channels/601130461678272522/1112829613920505956/1117440843289198643) of @kubouch 
> I think we're conflating two things here a bit:
> 1. Fetch the package from the internet
> 2. Install the package
>
> The step 2 would be identical for any project regardless of where it comes from. The difference between a git package and a local package would be in the step 1.

this means that this PR implements step 2!
so i assume here that the *package* is there and valid, hence the note and the missing checks.

**do you think we should add them here?**
yeah probably, because if you clone the project and then `nupm install` locally, if the *package* file check is only done in the firt stage, you might try to install a broken package :thinking:

Reply:
@Yethal 
i've added a check for the existence and the validity of the `package.nuon` file in e015a28 :+1: 

## examples
- when `package.nuon` does not exist
```bash
Error:   × package_file_not_found:
  │ no 'package.nuon' found in /home/amtoine/.local/share/git/store/github.com/amtoine/nu-git-manager
```
- when the file is missing required keys
```bash
Error:   × invalid_package_file:
  │ /home/amtoine/.local/share/git/store/github.com/amtoine/nu-git-manager/package.nuon is missing the following required
  │ keys: description, license
```